spring-kafka/src/main/java/org/springframework/kafka/core/DefaultShareConsumerFactory.java

```diff
@@ -0,0 +1,356 @@
+	}
+
+	/**
+	 * Set a supplier to supply instances of the key deserializer. The deserializer will
+	 * be configured using the consumer configuration, unless
+	 * {@link #setConfigureDeserializers(boolean) configureDeserializers} is false.
+	 * @param keyDeserializerSupplier the supplier (nullable).
+	 */
+	public void setKeyDeserializerSupplier(@Nullable Supplier<@Nullable Deserializer<K>> keyDeserializerSupplier) {
+		this.keyDeserializerSupplier = keyDeserializerSupplier;
+	}
+
+	/**
+	 * Set a supplier to supply instances of the value deserializer. The deserializer will
+	 * be configured using the consumer configuration, unless
+	 * {@link #setConfigureDeserializers(boolean) configureDeserializers} is false.
+	 * @param valueDeserializerSupplier the supplier (nullable).
+	 */
+	public void setValueDeserializerSupplier(@Nullable Supplier<@Nullable Deserializer<V>> valueDeserializerSupplier) {
+		this.valueDeserializerSupplier = valueDeserializerSupplier;
+	}
+
+	/**
+	 * Set to false (default true) to prevent programmatically provided deserializers (via
+	 * constructor or setters) from being configured using the consumer configuration,
+	 * e.g. if the deserializers are already fully configured.
+	 * @param configureDeserializers false to not configure.
+	 * @see #setKeyDeserializer(Deserializer)
+	 * @see #setKeyDeserializerSupplier(Supplier)
+	 * @see #setValueDeserializer(Deserializer)
+	 * @see #setValueDeserializerSupplier(Supplier)
+	 **/
+	public void setConfigureDeserializers(boolean configureDeserializers) {
+		this.configureDeserializers = configureDeserializers;
+	}
+
+	/**
+	 * Get the current list of listeners.
+	 * @return the listeners.
+	 */
+	@Override
+	public List<Listener<K, V>> getListeners() {
+		return Collections.unmodifiableList(this.listeners);
+	}
+
+	/**
+	 * Add a listener.
+	 * @param listener the listener.
+	 */
+	@Override
+	public void addListener(Listener<K, V> listener) {
+		Assert.notNull(listener, "'listener' cannot be null");
+		this.listeners.add(listener);
+	}
+
+	/**
+	 * Add a listener at a specific index.
+	 * <p>
+	 * This method allows insertion of a listener at a particular position in the internal listener list.
+	 * While this enables ordering of listener callbacks (which can be important for certain monitoring or extension scenarios),
+	 * there is intentionally no corresponding {@code removeListener(int index)} contract. Removing listeners by index is
+	 * discouraged because the position of a listener can change if others are added or removed, making it easy to
+	 * accidentally remove the wrong one. Managing listeners by their reference (object) is safer and less error-prone,
+	 * especially as listeners are usually set up once during initialization.
+	 * {@see #removeListener(Listener)}
+	 * </p>
+	 * @param index the index (list position).
+	 * @param listener the listener to add.
+	 * @since 2.5
```

Comment:
Probably we don't need this `@since` since the whole class is new in this `4.0` version.

Reply:
copy/paste error :)